test/celeritas/field/DormandPrinceStepper.test.cu

```diff
@@ -0,0 +1,318 @@
+    int *d_number_iterations, *d_number_threads;
+    arguments_size += sizeof(int) * 2;
+
+    FieldStepperResult *h_results, *d_results;
+    h_results = new FieldStepperResult[number_states];
+    for (int i = 0; i < number_states; ++i)
+    {
+        h_results[i] = FieldStepperResult();
+        arguments_size += sizeof(FieldStepperResult);
+    }
+
+    OdeState *h_along_state, *d_along_state;
+    if (number_threads > 1 && !use_shared)
+    {
+        h_along_state = new OdeState[number_states];
+        for (int i = 0; i < number_states; ++i)
+        {
+            h_along_state[i] = OdeState();
+            arguments_size += sizeof(OdeState);
+        }
+    }
+
+    OdeState *h_ks, *d_ks;
+    if (number_threads > 1 && !use_shared)
+    {
+        h_ks = new OdeState[number_states * 7];
+        for (int i = 0; i < number_states * 7; ++i)
+        {
+            h_ks[i] = OdeState();
+            arguments_size += sizeof(OdeState);
+        }
+    }
+
+    OdeState *h_states, *d_states;
+    h_states = new OdeState[number_states];
+    for (int i = 0; i < number_states; ++i)
+    {
+        h_states[i] = initial_states[i % number_states_sample];
+        arguments_size += sizeof(OdeState);
+    }
+
+    // Create events
+    cudaEvent_t start, stop;
+    cudaEventCreate(&start);
+    cudaEventCreate(&stop);
+
+    // Allocate memory on device
+    cudaMalloc(&d_results, number_states * sizeof(FieldStepperResult));
+    cudaMalloc(&d_states, number_states * sizeof(OdeState));
+    cudaMalloc(&d_number_iterations, sizeof(int));
+    cudaMalloc(&d_number_threads, sizeof(int));
+    if (number_threads > 1 && !use_shared)
+    {
+        cudaMalloc(&d_ks, number_states * 7 * sizeof(OdeState));
+        cudaMalloc(&d_along_state, number_states * sizeof(OdeState));
+    }
+
+    // Copy initial states to device
+    cudaMemcpy(d_states,
+               initial_states,
+               number_states * sizeof(OdeState),
+               cudaMemcpyHostToDevice);
+    cudaMemcpy(d_number_iterations,
+               &number_iterations,
+               sizeof(int),
+               cudaMemcpyHostToDevice);
+    cudaMemcpy(
+        d_number_threads, &number_threads, sizeof(int), cudaMemcpyHostToDevice);
+
+    if (number_threads > 1 && !use_shared)
+    {
+        cudaMemcpy(d_ks,
+                   h_ks,
+                   number_states * 7 * sizeof(OdeState),
+                   cudaMemcpyHostToDevice);
+        cudaMemcpy(d_along_state,
+                   &h_along_state,
+                   number_states * sizeof(OdeState),
+                   cudaMemcpyHostToDevice);
+    }
+
+    // Kernel configuration
+    int shared_memory = 0;
+    if (use_shared)
+    {
+        shared_memory = number_states * 7 * sizeof(OdeState)
+                        + number_states * sizeof(OdeState)
+                        + number_states * sizeof(FieldStepperResult);
+    }
+    int block_dimension = 1;
+    int thread_dimension = number_threads * number_states;
```

Comment:
maximum across CUDA architecture is 1024, in practice probably less because of registers usage. Maybe have a fixed block size and vary the number of blocks? It also makes it easier for the shared memory case; it won't grow out of control over the physical device limit.